### PR TITLE
Feat/hls

### PR DESCRIPTION
**Pull request by HassanAlsamahi**
This PR for the addition of color conversion to HLS color space.

### CODE REVIEW

**Comment by HassanAlsamahi** [comment]
I can't understand the circleci log, why the tests fail?

**Comment by ducha-aiki** [comment]
Looks like smth updated with conda. We will take a look into CI issues

**Comment by HassanAlsamahi** [comment]
Okay great, thanks @ducha-aiki

File: kornia/color/hls.py [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+    def __init__(self) -> None:
+        super(HlsToRgb, self).__init__()
+
+    def forward(self, image: torch.Tensor) -> torch.Tensor:  # type: ignore
+        return hls_to_rgb(image)
+
+
+def hls_to_rgb(image):
+    r"""Convert an HSV image to RGB
```
Comment (edgarriba) [inline kornia/color/hls.py]: Hls

File: kornia/color/hls.py [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    The image data is assumed to be in the range of (0, 1).
+
+    Args:
+        input (torch.Tensor): RGB Image to be converted to HSV.
+
+
+    Returns:
+        torch.Tensor: HSV version of the image.
+    """
```
Comment (edgarriba) [inline kornia/color/hls.py]: Hls

File: kornia/color/hls.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+    if not torch.is_tensor(image):
+        raise TypeError("Input type is not a torch.Tensor. Got {}".format(
+            type(image)))
+
+    if len(image.shape) < 3 or image.shape[-3] != 3:
+        raise ValueError("Input size must have a shape of (*, 3, H, W). Got {}"
+                         .format(image.shape))
+
```
Comment (edgarriba) [inline kornia/color/hls.py]: Wondering whether we should check if input is in floating point precision. @ducha-aiki

File: kornia/color/hls.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    Imax: torch.Tensor = image.max(-3)[1]
+    Imin: torch.Tensor = image.min(-3)[1]
+
+    l: torch.Tensor = (maxc + minc) / 2  # luminance
+
+    deltac: torch.Tensor = maxc - minc
+    s: torch.Tensor = torch.zeros_like(deltac)
+    s[l < 0.5]: torch.Tensor = (deltac[l < 0.5]) / (maxc[l < 0.5] + minc[l < 0.5])
+    s[l >= 0.5]: torch.Tensor = (deltac[l >= 0.5]) / (2 - (maxc[l >= 0.5] + minc[l >= 0.5]))  # Saturation
```
Comment (edgarriba) [inline kornia/color/hls.py]: I would use here torch.where and just computer once `l < 0.5` and `l >= 0.5`

File: kornia/color/hls.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+
+    hi = torch.zeros_like(deltac)
+    hi[Imax == 0] = ((g[Imax == 0] - b[Imax == 0]) / deltac[Imax == 0]) % 6
+    hi[Imax == 1] = ((b[Imax == 1] - r[Imax == 1]) / deltac[Imax == 1]) + 2
+    hi[Imax == 2] = ((r[Imax == 2] - g[Imax == 2]) / deltac[Imax == 2]) + 4
```
Comment (edgarriba) [inline kornia/color/hls.py]: Same here, precompute and reuse masks

File: test/color/test_hls.py [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+        data = torch.tensor([[[21., 22.],
+                              [22., 22.]],
+
+                             [[13., 14.],
+                              [14., 14.]],
+
+                             [[8., 8.],
+                              [8., 8.]]])
+
+        expected = torch.tensor([[[0.0641, 0.07138],
```
Comment (edgarriba) [inline test/color/test_hls.py]: From where did you get this expected values. Worth mentioning it in a comment

Comment (HassanAlsamahi) [reply test/color/test_hls.py]: Okay great I'll check the changes and push a commit.
 I got the expected values from an online color convertor, I don't know is that okay or no? if yes do I add a comment line with the link of the website?

Comment (edgarriba) [reply test/color/test_hls.py]: ideally we should be able to provide a reference from where our implementation comes from. Then we can justify the unit tests in a more robust way :D

Comment (HassanAlsamahi) [reply test/color/test_hls.py]: pardon my confusion, can you clarify more about the reference thing, where or how to mention them :D

Comment (edgarriba) [reply test/color/test_hls.py]: you can add it as comment in rgb_to_hls or in the documentation. This is a not very common transform so at least a minim of documentation stating what's the algorithm behind that

**Comment by edgarriba** [comment]
@HassanAlsamahi thanks for the contribution ! let us know if you have any other contribution in mind. There are still few things related to color to do :D
See this: https://github.com/arraiyopensource/kornia/pull/172
and https://github.com/arraiyopensource/kornia/issues/256
and https://github.com/arraiyopensource/kornia/issues/239

**Comment by HassanAlsamahi** [comment]
@edgarriba  okay great, I'll work on them, thanks :D
